### ISSUE TITLE
Convert command is appending project name into resulting config

### ISSUE THREAD

**Issue by designermonkey** [docker/compose#9445]
When running the following command:

```
docker compose -f docker-compose.yml convert
```

Using the following docker-compose.yml file

```yaml
---
version: '3'
services:
  my-service:
    hostname: my-service.domain.net
    image: someimage:1.0
```

The resulting yaml has the property `name: foldername` at the root. As far as I was aware, this is an invalid property and should not be there. It errors in docker compose and errors in docker swarm (which is why I use `convert` in the first place.

If the docker compose file is in a folder called badger, the output is:

```yaml
name: badger
services:
  my-service:
    hostname: my-service.domain.net
    image: someimage:1.0
    networks:
      default: null
networks:
  default:
    name: badger_default
```

If the project name is manually set as per `docker compose -p mushroom -f docker-compose.yml convert` then the resulting yaml is:

```yaml
name: mushroom
services:
  my-service:
    hostname: my-service.domain.net
    image: someimage:1.0
    networks:
      default: null
networks:
  default:
    name: mushroom_default
```

I would expect that the project name is not appended to the yaml output.

**Output of `docker compose version`:**

```
Docker Compose version v2.4.1
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.8.2)
  compose: Docker Compose (Docker Inc., v2.4.1)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 48
  Running: 32
  Paused: 0
  Stopped: 16
 Images: 48
 Server Version: 20.10.14
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: active
  NodeID: rj0zxhne7x2cdw0q5trnyo51m
  Is Manager: true
  ClusterID: r3afy781yls4clg969kaei5nh
  Managers: 1
  Nodes: 1
  Default Address Pool: 10.0.0.0/8  
  SubnetSize: 24
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 0
  Autolock Managers: false
  Root Rotation In Progress: false
  Node Address: 192.168.65.3
  Manager Addresses:
   192.168.65.3:2377
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3df54a852345ae127d1fa3092b95168e4a88e2f8
 runc version: v1.0.3-0-gf46b6ba
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.104-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.765GiB
 Name: docker-desktop
 ID: MKY4:GYXB:UWNR:R7VV:4EY4:5WOH:7OGE:3ZIQ:IWIQ:FFKH:Q3MT:T7CW
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```

I hope this is sufficient information. This only started happening after I updated yesterday.

**Comment by ndeloof** [comment]
> As far as I was aware, this is an invalid property and should not be there

it is valid per the [Compose Specification](https://compose-spec.io/)

`docker compose config` command is not designed to be used as input for `docker swarm` command, as the latter doesn't support the compose specification. We recently introduced `docker swarm config` command to avoid this confusion

**Comment by designermonkey** [comment]
Ok I have never seen that spec before. Can it be explained why when using convert, the resulting yaml causes errors in both docker compose and docker stack?

**Comment by Erokos** [comment]
@ndeloof I'm sorry but I can't find any reference to the `docker swarm config` comand youmentioned.
 https://docs.docker.com/engine/reference/commandline/swarm/

**Comment by ndeloof** [comment]
my bad: `docker stack config`  https://docs.docker.com/engine/reference/commandline/stack_config/

**Comment by Erokos** [comment]
so, I've just updated my Docker Desktop version to the latest one available and when i try to list the stack subcommands, I still don't see the config command. Is it only available for EE?

**Comment by ndeloof** [comment]
This is available with docker CLI v23.0.1, recently release (will be available with incoming Docker Desktop 4.17 release)
